src/FormSerializationWriter.cs

```diff
@@ -162,18 +173,21 @@ public void WriteCollectionOfPrimitiveValues<T>(string? key, IEnumerable<T>? val
     {
         if(depth > 0) throw new InvalidOperationException("Form serialization does not support nested objects.");
         depth++;
-        if(value == null && !additionalValuesToMerge.Any(static x => x != null)) return;
+        if (value == null && !additionalValuesToMerge.GetEnumerator().MoveNext()) return;
         if(value != null) {
             OnBeforeObjectSerialization?.Invoke(value);
             OnStartObjectSerialization?.Invoke(value, this);
             value.Serialize(this);
         }
-        foreach(var additionalValueToMerge in additionalValuesToMerge.Where(static x => x != null))
+        foreach (var additionalValueToMerge in additionalValuesToMerge)
         {
-            OnBeforeObjectSerialization?.Invoke(additionalValueToMerge!);
-            OnStartObjectSerialization?.Invoke(additionalValueToMerge!, this);
-            additionalValueToMerge!.Serialize(this);
-            OnAfterObjectSerialization?.Invoke(additionalValueToMerge!);
+            if (additionalValueToMerge != null)
```

Comment:
nit, revert the condition, use continue, and remove the added curlies to avoid the arrow head shape of the code